File: Programming - 2/Lab_2/Lab2_CPP/CustomString.cpp
#include "CustomString.h"


	CustomString::CustomString() {
		_length = 0;
		_string = new char[1];
		_string[_length] = '\0';
	}

	CustomString::CustomString(char* charArray) {
		_length = 0;
		_string = new char[1];
		while (charArray[_length] != '\0') {
			_length++;
			delete[] _string;
			_string = new char[_length + 1];
			for (int i = 0; i < _length; i++) {
				_string[i] = charArray[i];
			}
		}
		_string[_length] = '\0';
	}

	CustomString::~CustomString() {
		//delete[] _string;
	}

	char* CustomString::GetString()
	{
		char* outString = new char[_length + 1];
		for (int i = 0; i < _length; i++)
			outString[i] = _string[i];
		outString[_length] = '\0';
		return outString;
		//return _string;
	}

	int CustomString::GetLength() {
		return _length;
	}

	void CustomString::Clear()
	{
		delete[] _string;
		_length = 0;
	}

File: Programming - 2/Lab_2/Words/CustomString.cs

namespace Words
{
    public class CustomString
    {
		public CustomString()
		{
			_length = 0;
			_string = new char[0];
		}

		public CustomString(char[] charArray)
		{
			_length = charArray.Length;
			_string = new char[_length];
			for (int i = 0; i < _length; i++)
			{
				_string[i] = charArray[i];
            }
		}

		public char[] GetString()
        {
            char[] outString = new char[_length];
            for (int i = 0; i < _length; i++)
                outString[i] = _string[i];
			return outString;
            //return _string;
        }
		public int GetLength()
		{
			return _length;
		}

		public void Clear()
        {
			_string = null;
			_length = 0;
        }
		

		private char[] _string;
		private int _length;
	}
}
